library/src/interactor_impl.cxx

```diff
@@ -167,9 +167,15 @@ class interactor_impl::internals
     bool checkColoring = false;
     bool render = false;
 
-    // Available keycodes: W
+    // Available keycodes: None
     switch (keyCode)
     {
+      case 'W':
+        self->AnimationManager->CycleAnimation();
+        self->Options.set("scene.animation.index", self->AnimationManager->GetAnimationIndex());
+        ren->SetAnimationnameInfo(self->AnimationManager->GetAnimationName());
```

Comment:
You need to add a dedicated method in window_impl.h and window_impl.cxx that calls ` ren->SetAnimationnameInfo(name);`

Call that method from `loader_impl::LoadGeometry` and `loader_impl::LoadScene` after `this->Internals->AnimationManager.Initialize`